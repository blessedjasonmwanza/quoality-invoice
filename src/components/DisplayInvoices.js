import React, { useEffect, useState } from 'react'
import { useSelector } from 'react-redux';
import '../assets/css/Table.css';

export default function DisplayInvoices({page, status}) {
  const invoices = useSelector(state => state.invoices.invoices);
  console.log(invoices);
  const [invoicesToDisplay, setInvoicesToDisplay] = useState([]);
  useEffect(() => {
    if(status.toLowerCase() === 'all') {
      setInvoicesToDisplay(invoices)
    }else{
      const invoicesList = invoices.filter(invoice => invoice.status === status);
      setInvoicesToDisplay(invoicesList)
    }
  },[]);
  const editInvoice = (invoice) => alert(`Edit ${invoice.company_name} invoice coming soon!`);
  return (
    <>
      <span className='display-title'>{page}</span>
      <div className='card-hovered'>
        <table id="table" className='dt-table'>
          <thead>
            <tr>
              <th>Company</th>
              <th>Items</th>
              <th>Subtotal</th>
              <th>Total discount</th>
              <th>Total Excluding Tax</th>
              <th>Actions</th>
            </tr>
          </thead>
          <tbody>
            {
              invoicesToDisplay.map((invoice, i) => (
                <tr key={i}>
                  <td>{invoice.company_name}</td>
                  <td>{invoice.invoice_items.map((item) => item.name).join(', ')}</td>
                  <td>{invoice.subtotal}</td>
                  <td>{invoice.total_discount}</td>
                  <td>{invoice.total_excl_tax}</td>
                  <td className='actions'>
                    <button className='btn' onClick={() => editInvoice(i)}>Edit</button>
                  </td>
                </tr>)
              )
            }
          </tbody>
        </table>
      </div>
    </>
  )
}
