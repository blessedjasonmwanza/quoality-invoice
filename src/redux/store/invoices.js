/* eslint-disable */
import { createSlice } from '@reduxjs/toolkit';
import saveInvoices from '../../components/orders/SaveInvoices';
// slice
const invoicesSlice = createSlice({
  name: 'invoices',
  initialState: {
    invoices: [],
  },
  reducers: {
    invoicesSuccess: (state, action) => {
      state.invoices = action.payload;
      saveInvoices(action.payload);
    },
    addInvoice: (state, action) => {
      const newInvoices = [...state.invoices, action.payload];
      saveInvoices(newInvoices);
      state.invoices = newInvoices;
    },
    searchResults: (state, action) => {
      state.invoices = action.payload;
    },
    updateInvoiceStatus: (state, action) => {
      const {id, status} = action.payload;
      const updatedInvoices = state.invoices.map(order => (order.id === id ? { ...order, status: status } : order));
      saveInvoices(updatedInvoices);
      state.invoices = updatedInvoices;
    }
  },
});

export default invoicesSlice.reducer;

// Actions
export const {
  invoicesSuccess, searchResults, updateInvoiceStatus, addInvoice,
} = invoicesSlice.actions;
