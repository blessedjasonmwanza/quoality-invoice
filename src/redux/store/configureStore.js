/* eslint-disable */
import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import invoices from './invoices';

export const reducer = combineReducers({
  // reducers go here
  invoices,
});

const store = configureStore({
  reducer,
  middleware: [thunk, logger],
});

export default store;
