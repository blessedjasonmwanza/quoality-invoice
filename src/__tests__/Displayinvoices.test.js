import React from 'react';
import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux'
import DisplayInvoices from '../components/DisplayInvoices'
import store from '../redux/store/configureStore';

let root;

describe('DisplayInvoices', () => {
  it('renders Overdue invoices page without crashing', () => {
    act((() => {
      root = create(
        <Provider store={store}>
          <DisplayInvoices page='Overdue Invoices' status='overdue' />
        </Provider>
      )
      console.log(root);
    }))
    expect(root.toJSON()).toMatchSnapshot();
  });

  it('renders Paid invoices page without crashing', () => {
    act((() => {
      root = create(
        <Provider store={store}>
          <DisplayInvoices page='Paid Invoices' status='paid' />
        </Provider>
      )
      console.log(root);
    }))
    expect(root.toJSON()).toMatchSnapshot();
  });
  it('renders Unpaid invoices page without crashing', () => {
    act((() => {
      root = create(
        <Provider store={store}>
          <DisplayInvoices page='Unpaid Invoices' status='unpaid' />
        </Provider>
      )
      console.log(root);
    }))
    expect(root.toJSON()).toMatchSnapshot();
  });
  it('renders All invoices page without crashing', () => {
    act((() => {
      root = create(
        <Provider store={store}>
          <DisplayInvoices page='All Invoices' status='all' />
        </Provider>
      )
      console.log(root);
    }))
    expect(root.toJSON()).toMatchSnapshot();
  });
});