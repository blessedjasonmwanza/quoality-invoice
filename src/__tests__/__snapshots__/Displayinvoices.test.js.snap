// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayInvoices renders All invoices page without crashing 1`] = `
Array [
  <span
    className="display-title"
  >
    All Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`DisplayInvoices renders Overdue invoices page without crashing 1`] = `
Array [
  <span
    className="display-title"
  >
    Overdue Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`DisplayInvoices renders Paid invoices page without crashing 1`] = `
Array [
  <span
    className="display-title"
  >
    Paid Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`DisplayInvoices renders Unpaid invoices page without crashing 1`] = `
Array [
  <span
    className="display-title"
  >
    Unpaid Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`With React Testing Library DisplayInvoices renders Overdue invoices page without crashing 1`] = `
Array [
  <span
    className="display-title"
  >
    Overdue Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`With React Testing Library DisplayInvoices renders Overdue invoices page without crashing 2`] = `
Array [
  <span
    className="display-title"
  >
    Overdue Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`With React Testing Library DisplayInvoices renders Overdue invoices page without crashing 3`] = `
Array [
  <span
    className="display-title"
  >
    Overdue Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`With React Testing Library DisplayInvoices renders Paid invoices page without crashing 1`] = `
Array [
  <span
    className="display-title"
  >
    Paid Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;

exports[`With React Testing Library renders Paid invoices page without crashing 1`] = `
Array [
  <span
    className="display-title"
  >
    Paid Invoices
  </span>,
  <div
    className="card-hovered"
  >
    <div
      className="invoice-search-bar d-flex"
    >
      <span />
      <label>
        <small>
          Find Invoices
        </small>
        <input
          onInput={[Function]}
          placeholder="Search"
          type="search"
        />
      </label>
    </div>
    <table
      className="dt-table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Items
          </th>
          <th>
            Subtotal
          </th>
          <th>
            Total discount
          </th>
          <th>
            Total Excluding Tax
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;
